### PR TITLE
[CI/Infra] Fix DevNet e2e tests post-migration with validators

### PR DESCRIPTION
**Pull request by okdas**
### Human Summary

Fixing e2e test infrastructure post 0.47.x -> 0.50.x migration.

This PR should be merged with https://github.com/pokt-network/helm-charts/pull/31/files so LocalNet continues working.

If you want to run this branch on LocalNet, make sure to have your `helm-charts` repo pulled locally and pointed to the `no-rollkit-args` branch.

### AI Summary

reviewpad:summary

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [x] **Run all unit tests**: `make go_develop_and_test`
- [x] **Run E2E tests locally**: `make test_e2e`
- [x] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [x] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

**Comment by okdas** [comment]
E2E tests work in a sense that they can run. However, I encounter an issue I'm not sure how to proceed with. Both locally and remotely (on DevNet) e2e tests get stuck when they try to send a relay. In particular, this command runs until it times out: 

```bash
curl -v -sS -X 'POST' -H 'Content-Type: application/json' --data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[],\"id\":1}" http://localhost:42069/anvil
```

The appgate-server has some logging output:
```
{"level":"info","listening_endpoint":"http://0.0.0.0:42069","message":"Starting AppGate server..."}
{"level":"info","endpoint":":9090","message":"serving metrics"}
{"level":"debug","service_id":"anvil","payload":"{\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[],\"id\":1}","message":"handling relay"}
{"level":"debug","message":"determining request type"}
{"level":"debug","payload":"{\"jsonrpc\":\"2.0\",\"method\":\"eth_blockNumber\",\"params\":[],\"id\":1}","message":"partially Unmarshalling request"}
{"level":"debug","message":"Validating basic JSON payload"}
{"level":"debug","request_type":"JSON_RPC","message":"got request type"}

***** CURL REQUEST WAITING HERE, THE PANIC APPEARS A MOMENT AFTER CURL IS STOPPED *****


{"level":"warn","error":"observer is closed","message":"redundant unsubscribe"}
2024/02/24 01:04:01 http: panic serving 127.0.0.1:58536: runtime error: index out of range [0] with length 0
goroutine 26 [running]:
net/http.(*conn).serve.func1()
    /opt/homebrew/Cellar/go/1.22.0/libexec/src/net/http/server.go:1898 +0x118
panic({0x3dc8520?, 0x40007b01e0?})
    /opt/homebrew/Cellar/go/1.22.0/libexec/src/runtime/panic.go:770 +0xf0
github.com/pokt-network/poktroll/pkg/sdk.(*poktrollSDK).GetSessionSupplierEndpoints(0x4001614500, {0x4e76d00, 0x4001832360}, {0x40012ade30, 0x2b}, {0x40007b0096, 0x5})
    /Users/dk/pocket/poktroll/pkg/sdk/session.go:45 +0x9d0
github.com/pokt-network/poktroll/pkg/appgateserver.(*appGateServer).handleSynchronousRelay(0x4001660400, {0x4e76d00, 0x4001832360}, {0x40012ade30, 0x2b}, {0x40007b0096, 0x5}, {0x4001734200, 0x3f, 0x200}, ...)
    /Users/dk/pocket/poktroll/pkg/appgateserver/synchronous.go:29 +0x2a8
github.com/pokt-network/poktroll/pkg/appgateserver.(*appGateServer).ServeHTTP(0x4001660400, {0x4e74680, 0x400165e060}, 0x4001d48120)
    /Users/dk/pocket/poktroll/pkg/appgateserver/server.go:220 +0x718
github.com/slok/go-http-metrics/middleware/std.Handler.func1.1()
    /Users/dk/gopath/pkg/mod/github.com/slok/go-http-metrics@v0.11.0/middleware/std/std.go:27 +0x58
github.com/slok/go-http-metrics/middleware.Middleware.Measure({{{0x4e74620, 0x4000a16108}, {0x0, 0x0}, 0x0, 0x0, 0x0}}, {0x0, 0x0}, {0x4e7c8a0, ...}, ...)
    /Users/dk/gopath/pkg/mod/github.com/slok/go-http-metrics@v0.11.0/middleware/middleware.go:117 +0x434
github.com/slok/go-http-metrics/middleware/std.Handler.func1({0x4e64b60, 0x40017a60e0}, 0x4001d48120)
    /Users/dk/gopath/pkg/mod/github.com/slok/go-http-metrics@v0.11.0/middleware/std/std.go:26 +0x274
net/http.HandlerFunc.ServeHTTP(0x400006f720, {0x4e64b60, 0x40017a60e0}, 0x4001d48120)
    /opt/homebrew/Cellar/go/1.22.0/libexec/src/net/http/server.go:2166 +0x40
net/http.serverHandler.ServeHTTP({0x40016af0e0}, {0x4e64b60, 0x40017a60e0}, 0x4001d48120)
    /opt/homebrew/Cellar/go/1.22.0/libexec/src/net/http/server.go:3137 +0x2a0
net/http.(*conn).serve(0x40016041b0, {0x4e76d38, 0x40013fe190})
    /opt/homebrew/Cellar/go/1.22.0/libexec/src/net/http/server.go:2039 +0x15f8
created by net/http.(*Server).Serve in goroutine 1
    /opt/homebrew/Cellar/go/1.22.0/libexec/src/net/http/server.go:3285 +0x88c

```

I am continuing to investigate this problem.

File: e2e/tests/session_steps_test.go [hunk 1]
```diff
@@ -213,3 +212,30 @@ func (s *suite) waitForMessageAction(action string) {
 		s.Log("Success; message detected before timeout.")
 	}
 }
+
+/*
+TODO_TECHDEBT/TODO_CONSIDERATION(#XXX): this duplicates the unexported block event
```
Comment (bryanchriswhite) [inline e2e/tests/session_steps_test.go:217]:
```suggestion
TODO_TECHDEBT/TODO_CONSIDERATION(#XXX): this duplicates the unexported tx event
```

File: e2e/tests/session_steps_test.go [hunk 2]
```diff
@@ -213,3 +212,30 @@ func (s *suite) waitForMessageAction(action string) {
...
+
+type from pkg/client/block/block.go. We seem to have some conflicting preferences
```
Comment (bryanchriswhite) [inline e2e/tests/session_steps_test.go:219]:
```suggestion
TODO_TECHDEBT/TODO_CONSIDERATION(#XXX): this duplicates the unexported tx event
type from pkg/client/tx/client.go. We seem to have some conflicting preferences
```

File: localnet/poktrolld/config/app.toml [hunk 3]
```diff
@@ -49,10 +49,13 @@ query-gas-limit = "0"
     stop-node-on-err = true
 
 [telemetry]
+  datadog-hostname = ""
```
Comment (Olshansk) [inline localnet/poktrolld/config/app.toml:52]: We're going with datadog?

Comment (okdas) [reply localnet/poktrolld/config/app.toml:52]: @Olshansk we don't - this is a default config file that was created by ignite.

**Review by Olshansk — APPROVED** [review]
Review commit here: [`597d6b1` (#398)](https://github.com/pokt-network/poktroll/pull/398/commits/597d6b1d27bee2d4d0d056b89a787c89f50cd968)

**Comment by okdas** [comment]
@Olshansk FYI - I've changed the name of the chart from `poktroll-validator1` to `poktroll-validator` [here](https://github.com/pokt-network/helm-charts/blob/main/charts/poktroll-validator/Chart.yaml#L2), which triggered some changes to this branch too. I've verified E2E pass locally. This should not affect the DevNet E2E tests, but let's wait for them to finish, too.

If you're curious why the change: the name of the chart should be the same as the name of the directory the chart is in. Reasons:
- that's how all community/OSS charts are structured. 
- in our LocalNet setup, when `helm_chart_local_repo.enabled` is `false` Tilt looks for the `poktroll-validator` chart in our helm chart repo. When the value is `true`, Tilt looks for the directory called `poktroll-validator` in the local git repo.

**Pull request merged**